crates/editor/src/editor.rs

```diff
@@ -5136,19 +5148,105 @@ impl Editor {
+        {
+            let info = excerpt_boundery.next.as_ref();
+
+            if let Some(info) = info {
+                let Some(excerpt_ranges) =
+                    multi_buffer_snapshot.range_for_excerpt::<Point>(info.id)
+                else {
+                    continue;
+                };
+
+                // To translate a breakpoint's position within a singular buffer to a multi buffer
+                // position we need to know it's excerpt starting location, it's position within
+                // the singular buffer, and if that position is within the excerpt's range.
+                let excerpt_head = excerpt_ranges
+                    .start
+                    .to_display_point(&snapshot.display_snapshot);
+                let buffer_range = info // Buffer lines being shown within the excerpt
+                    .buffer
+                    .summary_for_anchor::<Point>(&info.range.context.start)
+                    ..info
+                        .buffer
+                        .summary_for_anchor::<Point>(&info.range.context.end);
+
+                if let Some(breakpoints) = opened_breakpoints.get(&info.buffer_id) {
+                    for breakpoint in breakpoints {
+                        let breakpoint_position = info // Breakpoint's position within the singular buffer
+                            .buffer
+                            .summary_for_anchor::<Point>(&breakpoint.position.text_anchor);
+
+                        if buffer_range.contains(&breakpoint_position) {
+                            // Translated breakpoint position from singular buffer to multi buffer
+                            let delta = breakpoint_position.row - buffer_range.start.row;
+
+                            let position = excerpt_head + DisplayPoint::new(DisplayRow(delta), 0);
+
+                            breakpoint_display_points.insert(position);
+                        }
+                    }
+                };
+            };
+        }
+
+        breakpoint_display_points
+    }
+
     fn render_breakpoint(
         &self,
         position: Anchor,
         row: DisplayRow,
         cx: &mut ViewContext<Self>,
     ) -> IconButton {
+        let color = if self
+            .gutter_breakpoint_indicator
+            .is_some_and(|gutter_bp| gutter_bp.row() == row)
+        {
+            Color::Hint
+        } else {
+            Color::Debugger
+        };
+
         IconButton::new(("breakpoint_indicator", row.0 as usize), ui::IconName::Play)
             .icon_size(IconSize::XSmall)
             .size(ui::ButtonSize::None)
-            .icon_color(Color::Error)
+            .icon_color(color)
             .on_click(cx.listener(move |editor, _e, cx| {
                 editor.focus(cx);
-                editor.toggle_breakpoint_at_row(position, cx) //TODO handle folded
+                editor.toggle_breakpoint_at_anchor(position, cx) //TODO handle folded
```

Comment:
I think this todo can be removed, we already handle this case:
https://github.com/RemcoSmitsDev/zed/pull/13/files#diff-2c4beb0f3febb4181a2f3bcf1d1008cd789d2f072b34b57c3de3249d88ffe48fR1590

Reply:
I was thinking we could color a toggle the debugger accent color if there's a breakpoint nested within it. That's still not completed